functionary/sglang_monkey_patch/tokenizer_manager.py

```diff
@@ -0,0 +1,102 @@
+"""
+Copyright 2023-2024 SGLang Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""TokenizerManager is a process that tokenizes the text."""
+
+import asyncio
+import logging
+from typing import Dict, List, Optional, Tuple, Union
+
+import fastapi
+import uvloop
+from sglang.srt.managers.io_struct import (
+    EmbeddingReqInput,
+    GenerateReqInput,
+    RewardReqInput,
+)
+from sglang.srt.managers.tokenizer_manager import ReqState, TokenizerManager
+from sglang.srt.server_args import PortArgs, ServerArgs
+
+asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
+
+logger = logging.getLogger("tokenizer_logger")
+
+
+class MonkeyPatchTokenizerManager(TokenizerManager):
+    """TokenizerManager is a process that tokenizes the text."""
+
+    def __init__(
+        self,
+        server_args: ServerArgs,
+        port_args: PortArgs,
+        logfile: str = "logfile.txt",
```

Comment:
I just wonder we implement logfile for sglang, should we also implement for vllm? both should be similar in terms of features?